### DISCUSSION TITLE
OpenBSD support ?

### DISCUSSION THREAD

**Discussion by joyboy67** [aristocratos/btop#562] [General]
Hello, btop is available for OpenBSD (and NetBSD) ?

**Comment by deckstose** [comment]
No, not for now. Only FreeBSD works at them moment.

**Comment by joske** [comment]
How different is the system call API between FreeBSD and OpenBSD? It might be not so much work, but I don't have an OpenBSD install.

**Comment by joske** [comment]
I've done a bit of research and this seems like a fun project. Need to install openBSD on a VM or real HW first though.

**Comment by joske** [comment]
Installed OpenBSD 7.3 in QEMU (arm64 on M1 MBA)

**Comment by joske** [comment]
@joyboy67 Can you give the output of `sysctl hw.sensors` on a real openbsd system? On a VM there are no sensors.

**Comment by joske** [comment]
Adventurous people may try to compile this branch: https://github.com/joske/btop/tree/openbsd

You need to install g++ 11, git and gmake.

**Comment by joske** [comment]
@aristocratos On the above branch, most info is now available (only a few details missing). A blocker issue is that keys don't work. For some reason, `cin.get()` never returns a keypress. I haven't found out why that is yet. Maybe you can have a look? 

It's quite easy to install OpenBSD in a VM: download [this ISO](https://cloudflare.cdn.openbsd.org/pub/OpenBSD/snapshots/amd64/install73.iso) and install from it.

**Comment by joske** [comment]
@aristocratos never mind, fixed. Apparently the problem is 

```
				cin.sync_with_stdio(false);
				cout.sync_with_stdio(false);
```
